tests/utils.py

```diff
@@ -15,3 +16,10 @@ def upload_records(client: Client, num: int):
         record = client.create_record(data=data)
         records.append(record["data"])
     return records
+
+
+def create_extra_headers(username: str, password: str):
```

Comment:
This is due to a quirk of playwright. If you give playwright credentials to hit an endpoint, it _doesn't use them on the first request_. It expects a 401 response to come back, and then it will retry with credentials.

This doesn't work for us because our endpoints are publicly available in many cases. So rather than returning 401 they'll return only the publicly available information instead.

This method allows us to force an auth header to be present when we need it.